### PR TITLE
Upgrade to OmniSharp version 1.38.0

### PR DESCRIPTION
**Pull request by JoeRobich**

### CODE REVIEW

File: tasks/offlinePackagingTasks.ts [hunk 1]
```diff
@@ -24,13 +24,13 @@ import { getAbsolutePathPackagesToInstall } from '../src/packageManager/getAbsol
...
 
 export const offlinePackages = [
-    { platformInfo: new PlatformInformation('win32', 'x86_64'), id: "win32-x64" },
-    { platformInfo: new PlatformInformation('win32', 'x86'), id: "win32-ia32" },
-    { platformInfo: new PlatformInformation('win32', 'arm64'), id: "win32-arm64" },
-    { platformInfo: new PlatformInformation('linux', 'x86_64'), id: "linux-x64" },
-    { platformInfo: new PlatformInformation('linux', 'arm64'), id: "linux-arm64" },
-    { platformInfo: new PlatformInformation('darwin', 'x86_64'), id: "darwin-x64" },
-    { platformInfo: new PlatformInformation('darwin', 'arm64'), id: "darwin-arm64" },
+    { platformInfo: new PlatformInformation('win32', 'x86_64'), id: "win32-x64", isFramework: true },
```
Comment (JoeRobich) [inline tasks/offlinePackagingTasks.ts:27]: Ensure we are only packaging the Full Framework version of O#. Packaging both in addition to our debugger and razor lsp may push our size over the marketplace limit. We should transition in a couple releases to packaging the SDK build of O# and enabling it by default.

File: src/packageManager/downloadAndInstallPackages.ts [hunk 2]
```diff
@@ -16,15 +16,21 @@ import { mkdirpSync } from "fs-extra";
...
     if (packages) {
         eventStream.post(new PackageInstallStart());
         for (let pkg of packages) {
             let installationStage = "touchBeginFile";
 
-            if (pkg.url === null && pkg.id === "OmniSharp") {
-                eventStream.post(new InstallationFailure(installationStage, new Error("A release package of OmniSharp does not exist for this platform. Set \"omnisharp.path\" to \"latest\" in Settings to use an experimental build.")));
-                continue;
+            if (pkg.id === "OmniSharp") {
+                if (pkg.isFramework !== useFramework) {
```
Comment (JoeRobich) [inline src/packageManager/downloadAndInstallPackages.ts:26]: Ensure we are only looking at OmniSharp dependencies that match our framework option.

File: package.json [hunk 3]
```diff
@@ -565,16 +559,7 @@
...
-    "workspaceContains:*.sln",
-    "workspaceContains:*.slnf",
-    "workspaceContains:*.csx",
-    "workspaceContains:*.cake",
-    "workspaceContains:**/*.csproj",
-    "workspaceContains:**/*.sln",
-    "workspaceContains:**/*.slnf",
-    "workspaceContains:**/*.csx",
-    "workspaceContains:**/*.cake"
+    "workspaceContains:**/*.{csproj,sln,slnf,csx,cake}"
```
Comment (JoeRobich) [inline package.json:562]: Simplify our search for files to activate against.

File: package.json [hunk 4]
```diff
@@ -125,170 +125,154 @@
...
+      "integrity": "1E8B8EC8A69A5897017732442B90849393CA3FD2E5C33A406E25D8B065E5637E"
     },
     {
       "id": "OmniSharp",
       "description": "OmniSharp for Windows (.NET 6 / x86)",
-      "url": null,
-      "fallbackUrl": null,
-      "installPath": null,
+      "url": "https://roslynomnisharp.blob.core.windows.net/releases/1.38.0/omnisharp-win-x86-net6.0-1.38.0.zip",
+      "installPath": ".omnisharp/1.38.0-net6.0",
```
Comment (JoeRobich) [inline package.json:146]: This install path matches the path we use when downloading latest package and avoids conflicting with full framework installs.

**Review by filipw — APPROVED** [review]

**Pull request merged**